runtime/runtime/src/verifier.rs

```diff
@@ -122,11 +122,24 @@ pub fn validate_transaction(
     .map_err(InvalidTxError::ActionsValidation)
 }
 
-/// Verifies the signed transaction on top of given state, charges transaction fees
-/// and balances, and updates the state for the used account and access keys.
-pub fn verify_and_charge_transaction(
-    config: &RuntimeConfig,
+pub fn commit_charging_for_tx(
     state_update: &mut TrieUpdate,
+    tx: &Transaction,
+    signer: &Account,
+    access_key: &AccessKey,
+) {
+    set_access_key(state_update, tx.signer_id().clone(), tx.public_key().clone(), &access_key);
+    set_account(state_update, tx.signer_id().clone(), &signer);
+}
+
+/// Verifies the signed transaction on top of the given state; looks up the
+/// signer account and access_key from the transaction; updates them to charge
+/// for the transaction processing; returns the updated signer and access_key to
+/// the caller.  The caller can use `commit_charging_for_tx()` to commit the
+/// actual charging.
+pub fn verify_and_charge_tx_ephemeral(
```

Comment:
As an outsider, the name is slightly confusing. Without reading the description of `verify_and_charge_tx_ephemeral`, I would wonder are we really charging the transaction? What exactly does the ephemeral part mean here?

Would it be valid to name this function just `verify_tx`? We now have `commit_charging_for_tx` which makes it explicit that we need to commit the changes later.